src/components/Listing/ListingFilter.jsx

```diff
@@ -0,0 +1,167 @@
+                        onClick={() => dispatch({ type: "sort", payload: "men-clothing" })}
+                    />
+                    <label className="category" htmlFor="men">Men Clothing</label>
+                </p>
+                <p>
+                    <input className="category" type="radio" id="footwear" name="category" value="footwear"
+                        checked={state.sortBy === "footwear"}
+                        onClick={() => dispatch({ type: "sort", payload: "footwear" })}
+                    />
+                    <label className="category" htmlFor="footwear">footwear</label>
+                </p>
+                <p>
+                    <input className="category" type="radio" id="bags" name="category" value="bags"
+                        checked={state.sortBy === "bagpack"}
+                        onClick={() => dispatch({ type: "sort", payload: "bagpack" })}
+                    />
+                    <label className="category" htmlFor="kids">BagPacks</label>
+                </p>
+                <p>
+                    <input className="category" type="radio" id="electronics" name="category" value="electronics"
+                        checked={state.sortBy === "electronics"}
+                        onClick={() => dispatch({ type: "sort", payload: "electronics" })}
+                    />
+                    <label className="category" htmlFor="electronics">Electronics</label>
+                </p>
+
+            </div>
+            <div className="price-selector">
+                <div className="h5">Price</div>
+                <p>
+                    <input type="radio"
+                        checked={state.priceSelector === 499}
+                        onClick={() => dispatch({ type: "price", payload: 499 })}
+                    />
+                    <label className="price-container">499+</label>
+                </p>
+                <p>
+                    <input type="radio"
+                        checked={state.priceSelector === 1999}
+                        onClick={() => dispatch({ type: "price", payload: 1999 })}
+                    />
+                    <label className="price-container">1999+</label>
+                </p>
+                <p>
+                    <input type="radio"
+                        checked={state.priceSelector === 3599}
+                        onClick={() => dispatch({ type: "price", payload: 3599 })}
+                    />
+                    <label className="price-container">3599+</label>
+                </p>
+                <p>
+                    <input type="radio"
+                        checked={state.priceSelector === 4000}
+                        onClick={() => dispatch({ type: "price", payload: 4000 })}
+                    />
+                    <label className="price-container">4000+</label>
+                </p>
+            </div>
+            <div className="rating-selector">
```

Comment:
You can take an array of 5 and then map over it to return the rating block.
This will make you write less code. Moreover, this will make adding or deleting easier too.